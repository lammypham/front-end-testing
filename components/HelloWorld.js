import React from 'react'
import PositiveCounter from './PositiveCounter/PositiveCounter';

const HelloWorld = () => {

  return (
    <div className='hello-world'>
      <h1>Hello World</h1>
      <p>Welcome to my world</p>
      <span>I am testing</span>

      <PositiveCounter />
    </div>
  )
}
export default HelloWorld