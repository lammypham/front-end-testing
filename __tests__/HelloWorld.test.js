import HelloWorld from '../components/HelloWorld';
import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

test('HellWorld component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HelloWorld />, div);
});


it('renders a snapshot', () => {
    const tree = renderer.create(<HelloWorld />).toJSON();
    expect(tree).toMatchSnapshot();
})