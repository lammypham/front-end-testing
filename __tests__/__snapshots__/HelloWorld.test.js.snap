// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="hello-world"
>
  <h1>
    Hello World
  </h1>
  <p>
    Welcome to my world
  </p>
  <span>
    I am testing
  </span>
  <div>
    Value: 
    0
    <button
      className="decrement"
      onClick={[Function]}
    >
      âˆ’
    </button>
    <button
      className="increment"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</div>
`;
